### ISSUE TITLE
Provide a way to forcefully terminate a job.

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#994]
Originally reported of Google Code with ID 938.

```
Since Ganeti 2.12+ has jobs as processes, users might want to really *kill* a process (at they own risk). Ganeti should be able to send KILL (most likely as `gnt-job --yes-do-it ...`), and perhaps also QUIT, which would gave the process a change to clean up any local resources it might be using.

Also include the process ID in the list of jobs so that users can target the processes themselves.
```
Originally added on 2014-09-10 11:52:32 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
It would also be useful to give users the ability to check for dead jobs manually. Or at least, trigger such a check after sending a forceful termination signal to a running job.
```
Originally added on 2014-09-10 11:58:06 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 5e67cc2c7eb81612c03267a90c8cf08cdf228196
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Sep 24 11:48:21 2014 +0200

    After killing check for death
    
    Once we sent a sigKILL to a process there is a high chance that it
    actually died. So this is a good point in time to verify death and
    clean up the job queue.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 712f8a8a7ae2fe1e6d17a0fce7c01c49e48802a2
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Sep 24 12:01:11 2014 +0200

    Export function for job-death detection
    
    ...so that LuxiD can trigger that test directly, if it has reasons
    to believe a job died (e.g., after sending it a KILL signal).
    
    The function is a wrapper around checkForDeath that looks up the
    job by its id in the list of running jobs.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 2602dd1f95c71e380dedd408a1a0417b85be7ce2
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Sep 24 12:44:40 2014 +0200

    Guard --kill by --yes-do-it
    
    Forcefully killing a job has high potential of messing up
    parts of the cluster. So better double-check the user's intent.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 0702d0e4ae25d3f1536d529e27a953b6acb12dbd
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Sep 24 12:36:14 2014 +0200

    Document the --kill option in gnt-job cancel
    
    Also add a warning, that this option is a good way to mess
    up any objects touched by the job.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit d0135eb81b0b2894c1f37fd3b63eb432102293e4
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Sep 24 10:36:03 2014 +0200

    Add a kill option to gnt-job cancel
    
    Add a new --kill option to gnt-job cancel. This option
    will expose the new "kill" option of the CancelJob luxi
    command.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 87e5927fafaee6aebf5bdbb6ed7e9fa1b7966e1b
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Sep 24 10:42:45 2014 +0200

    Expose the optional kill in CancelJob in the luxi client
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 9ef96cec60916b830011bd515f4dc58fa01fce15
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Sep 24 11:21:11 2014 +0200

    Handle the kill option in CancelJob luxi requests
    
    To do so, extend the cancelJob function by an additional kill flag.
    Also, if we kill a job, we expect it to die regardless of whether it
    already acquired all the needed locks.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

commit 2e41a7c199414558e80ba86b332c4f9278496c8e
Author: Klaus Aehlig <aehlig@google.com>
Date:   Tue Sep 23 17:50:01 2014 +0200

    Extend the cancel-job luxi request by a kill flag
    
    This flag will instruct luxid to send send a SIGKILL
    instead of a SIGTERM if the job to be canceled has
    already started. To not have to change RAPI all in
    one big change, we keep the wire protocol backwards
    compatible even though this is not strictly necessary
    for the luxi protocol.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2014-09-24 15:30:40 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:37:03 +0000 UTC.

Changed State: Released
